src/common-components/Logistration.jsx

```diff
@@ -16,15 +16,21 @@ import { Redirect } from 'react-router-dom';
 
 import BaseComponent from '../base-component';
 import { LOGIN_PAGE, REGISTER_PAGE } from '../data/constants';
-import { getTpaHint, updatePathWithQueryParams } from '../data/utils';
+import { getTpaHint, getTpaProvider, updatePathWithQueryParams } from '../data/utils';
 import { LoginPage } from '../login';
 import { RegistrationPage } from '../register';
 import { backupRegistrationForm } from '../register/data/actions';
+import {
+  tpaProvidersAndSecondaryProvidersSelector,
+} from './data/selectors';
 import messages from './messages';
 
 const Logistration = (props) => {
-  const { intl, selectedPage } = props;
-  const tpa = getTpaHint();
+  const { intl, selectedPage, tpaProvidersAndSecondaryProviders } = props;
+  const tpaHint = getTpaHint();
+  const {
+    providers, secondaryProviders,
+  } = tpaProvidersAndSecondaryProviders;
```

Comment:
Do we have the providers when we first load the logistration? What if the call to to MFEContext took time, will it still work?

Reply:
I have provided default values for providers and secondary providers as empty list in common-components reducer, now suppose the logistration component is loaded and the MFEContext response is not received yet, in this case the `isValidTpaHint()` function will return false and the header will be displayed, and when the MFEContext response is received, the form will be rendered.